src/test/java/org/jenkinsci/plugins/workflow/test/steps/SemaphoreStep.java

```diff
@@ -255,5 +255,38 @@ public static class Execution extends AbstractStepExecutionImpl {
 
     }
 
+    // Marker interface just for clarity.
+    private interface KeyState { }
+
+    /**
+     * The step has not yet started, but will succeed immediately when it does start.
+     */
+    private static class ImmediateSuccessState implements KeyState {
```

Comment:
Clearer IMO to just call it a success state. The timing of that success is a matter for the code moving it around in the map.

Reply:
I think it is important to be able to distinguish "the step will succeed, when/if it starts" from "the step ran and finished". Maybe not for the common case, but for edge cases it seems important, and the old code was able to distinguish these states. Either way, I adjusted things in [d05eafd](https://github.com/jenkinsci/workflow-support-plugin/pull/260/commits/d05eafd27820d48db69890d04658053da4887743).